tests/test_other.py

```diff
@@ -9173,7 +9176,7 @@ def test_proxy_to_pthread_stack(self):
     # DEFAULT_PTHREAD_STACK_SIZE.
     self.do_smart_test(test_file('other/test_proxy_to_pthread_stack.c'),
                        ['success'],
-                       emcc_args=['-s', 'USE_PTHREADS', '-s', 'PROXY_TO_PTHREAD', '-s', 'DEFAULT_PTHREAD_STACK_SIZE=64kb', '-s', 'TOTAL_STACK=128kb'])
+                       emcc_args=['-s', 'USE_PTHREADS', '-s', 'PROXY_TO_PTHREAD', '-s', 'DEFAULT_PTHREAD_STACK_SIZE=64kb', '-s', 'TOTAL_STACK=128kb', '-s', 'EXIT_RUNTIME'])
```

Comment:
why is this needed?

Reply:
This is a little complicated.

node was previously not honoring `EXIT_RUNTIME=0` (the default).   We were forcing an exit via `proces.exit` from `quit_` (itself called from from `exit`) even when we want to runtime to to stay alive.    With this change, the modified `quit_` will now to the right thing and just throw/unwind rather than `process.exit` giving the process a chance to stay alive after `exit`.

When callMain() catches the ExitStatus() is simply ignores it (its just signal to unwind and be done).   If the node event loop has no more work to do the entire process will then exit.. simply by coincidence not because emscripten asked for an explicit exit.   This means that the behaviour of `EXIT_RUNTIME=0` and `EXIT_RUNTIME=1` were (and continue to be) kind of the same under node for simple programs that are not doing other async work.   This is why we don't need `EXIT_RUNTIME=1` on *all* our node unit tests, because node will naturally exit in most cases.

The situation for threaded programs is a little different:  As soon as any thread is started (or added to the thread pool) I believe we prevent the node process from naturally exiting since it needs to stick around to talk communicate with the thread.  

This means that threaded programs under node will not naturally exit and now require `-s EXIT_RUNTIME=1` to exit at all.     Does this seems unreasonable?   Should we need another setting such as `-s PROCESS_EXIT_ON_EXIT` which defaults to true independently of `EXIT_RUNTIME`?

Reply:
That node needs `EXIT_RUNTIME` in pthreads builds doesn't seem too bad to me, if we document it in the changelog and maybe also pthreads.rst. Node pthreads is still somewhat experimental I'd say. I don't think we need a new setting.